frontend/packages/dev-console/src/utils/__tests__/add-resources-menu-utils.spec.ts

```diff
@@ -0,0 +1,131 @@
+import { URL } from 'url';
+import { getMenuPath, getAddPageUrl } from '../add-resources-menu-utils';
+import {
+  fromGit,
+  fromCatalog,
+  containerImage,
+  fromDockerfile,
+  database,
+  addResourceMenu,
+} from '../../actions/add-resources';
+import {
+  transformTopologyData,
+  getTopologyResourceObject,
+} from '../../components/topology/topology-utils';
+import { ImportOptions } from '../../components/import/import-types';
+import { MockResources } from '../../components/topology/__tests__/topology-test-data';
+import { TopologyDataResources } from '../../components/topology/topology-types';
+
+const getTopologyData = (mockData: TopologyDataResources, transformByProp: string[]) => {
+  const result = transformTopologyData(mockData, transformByProp);
+  const keys = Object.keys(result.topology);
+  const resource = getTopologyResourceObject(result.topology[keys[0]]);
+  return { resource };
+};
+
+describe('addResourceMenuUtils: ', () => {
+  it('should give proper menu item path based on the application', () => {
+    let menuTitle = '';
+    menuTitle = getMenuPath(true);
+    expect(menuTitle).toEqual('Add to Application');
+
+    menuTitle = getMenuPath(false);
+    expect(menuTitle).toEqual('Add to Project');
+  });
+
+  it('should return the list of menu items for add resources', () => {
+    expect(addResourceMenu).toEqual([
+      fromGit,
+      containerImage,
+      fromCatalog,
+      fromDockerfile,
+      database,
+    ]);
+  });
+
+  it('should return the page url with proper queryparams for git import flow', () => {
+    const { resource } = getTopologyData(MockResources, ['deployments']);
+    const url = new URL(getAddPageUrl(resource, ImportOptions.GIT, true), 'https://mock.test.com');
+    // Check that url contains all expected params
+    expect(url.pathname).toBe('/import/ns/testproject1');
+    expect(url.searchParams.get('importType')).toBe('git');
+    expect(url.searchParams.get('application')).toBe('application-1');
+    expect(url.searchParams.get('isKnativeDisabled')).toBe('true');
+    // Negative checks for queryparams params.
+    expect(url.searchParams.has('preselected-ns')).toBe(false);
+    expect(url.searchParams.has('catagory')).toBe(false);
+  });
+
+  it('should return the page url without application params in the url', () => {
+    const { resource } = getTopologyData(MockResources, ['deployments']);
+    const url = new URL(getAddPageUrl(resource, ImportOptions.GIT, false), 'https://mock.test.com');
+    // Negative checks for queryparams params.
+    expect(url.searchParams.has('application')).toBe(false);
+    expect(url.searchParams.has('preselected-ns')).toBe(false);
+    expect(url.searchParams.has('catagory')).toBe(false);
```

Comment:
I think we can drop these two asserts... Really all we are looking for on this test is that you don't get an application param.